src/Simulation/CsharpGeneration/SimulationCode.fs

```diff
@@ -30,77 +30,26 @@ open Microsoft.Quantum.QsCompiler.Transformations.BasicTransformations
-
-        static member Apply (syntaxTree : IEnumerable<QsNamespace>) = 
-            let walker = new DeclarationPositions()
-            for ns in syntaxTree do 
-                walker.Transform ns |> ignore
-            walker.DeclarationLocations
-
-        override this.onSourceFile file = 
-            currentSource <- file.Value
-            file
-
-        override this.onLocation (loc : QsLocation) = 
-            if currentSource <> null then
-                declarationLocations.Add (NonNullable<string>.New currentSource, loc.Offset)
-            loc
-
-    type CodegenContext = { 
-        allQsElements           : IEnumerable<QsNamespace>
-        allUdts                 : ImmutableDictionary<QsQualifiedName,QsCustomType>
-        allCallables            : ImmutableDictionary<QsQualifiedName,QsCallable>
-        declarationPositions    : ImmutableDictionary<NonNullable<string>, ImmutableSortedSet<int * int>>
-        byName                  : ImmutableDictionary<NonNullable<string>,(NonNullable<string>*QsCallable) list>
-        current                 : QsQualifiedName option
-        signature               : ResolvedSignature option
-        fileName                : string option
-    } 
     
     type CodegenContext with
         member this.setCallable (op: QsCallable) = { this with current = (Some op.FullName); signature = (Some op.Signature) }
         member this.setUdt (udt: QsCustomType) = { this with current = (Some udt.FullName) } 
-
-    let createContext fileName syntaxTree =        
-        let udts = GlobalTypeResolutions syntaxTree
-        let callables = GlobalCallableResolutions syntaxTree
-        let positionInfos = DeclarationPositions.Apply syntaxTree
-        let callablesByName = 
-            let result = new Dictionary<NonNullable<string>,(NonNullable<string>*QsCallable) list>()
-            syntaxTree |> Seq.collect (fun ns -> ns.Elements |> Seq.choose (function
-            | QsCallable c -> Some (ns, c)
-            | _ -> None))
-            |> Seq.iter (fun (ns:QsNamespace,c:QsCallable) -> 
-                if result.ContainsKey c.FullName.Name then result.[c.FullName.Name] <- (ns.Name, c) :: (result.[c.FullName.Name]) 
-                else result.[c.FullName.Name] <- [ns.Name, c])
-            result.ToImmutableDictionary()
-        { 
-            allQsElements = syntaxTree; 
-            byName = callablesByName; 
-            allUdts = udts; 
-            allCallables = callables; 
-            declarationPositions = positionInfos.ToImmutableDictionary((fun g -> g.Key), (fun g -> g.ToImmutableSortedSet()))
-            current = None; 
-            fileName = fileName; 
-            signature = None 
-        }
               
-    let autoNamespaces = 
-        [
-            "System"
-            "Microsoft.Quantum.Core"
-            "Microsoft.Quantum.Intrinsic"
-            "Microsoft.Quantum.Simulation.Core" 
-        ]
+    let unitTestingNamespaces = [
```

Comment:
As mentioned before, I would not add these to the top, but simply use FQN for whatevery types it needs.